### PR TITLE
Implement email/password login with firebase

### PR DESCRIPTION
**Pull request by Anthony-Iozzia**
- login activity ( e-mail password)
- sign up activity (first name, last name, email, password)
- create user in firestore, get user in firestore
- display correct user info in HomeActivity

### CODE REVIEW

File: app/src/main/java/ch/epfl/qedit/view/login/Util.java [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+        languageSelectionSpinner.setSelection(positionInLanguageList, false);
+        // Set listener
+        languageSelectionSpinner.setOnItemSelectedListener(onItemSelectedListener);
+        // Set page title to display it in the right language
+        activity.setTitle(R.string.title_activity_sign_up);
+    }
+
+    /**
+     * Update the redirection text with colors (we cannot directly put the color in the strings.xml
+     * because we are changing language the language dynamically in these activities
```
Comment (brunnerant) [inline app/src/main/java/ch/epfl/qedit/view/login/Util.java:98]: There is a small typo in this comment ;).

File: app/src/main/java/ch/epfl/qedit/view/login/Util.java [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+    /**
+     * Put a string in the preferences.
+     *
+     * @param activity current activity
+     * @param key key to put
+     * @param value value to put with the key
+     */
+    @SuppressWarnings("SameParameterValue")
+    static void putStringInPrefs(Activity activity, String key, String value) {
+        SharedPreferences prefs = activity.getSharedPreferences("UserData", MODE_PRIVATE);
```
Comment (brunnerant) [inline app/src/main/java/ch/epfl/qedit/view/login/Util.java:156]: I would suggest that we create a `static final` attribute for `"UserData"`.

File: app/src/main/java/ch/epfl/qedit/view/login/LogInActivity.java [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+                            if (throwable != null) {
+                                Util.showToast(R.string.database_error, context, resources);
+                            } else {
+                                bundle.putSerializable(USER, user);
+                                intent.putExtras(bundle);
+                                startActivity(intent);
+                            }
+                        });
+        // Put the current user id in cache
+        Util.putStringInPrefs(this, "user_id", userId);
```
Comment (brunnerant) [inline app/src/main/java/ch/epfl/qedit/view/login/LogInActivity.java:166]: I suggest that we create a `static final` attribute in `login/Util.java` for `"user_id"`, same for the sign up counterpart.

File: app/src/androidTest/java/ch/epfl/qedit/login/SignUpActivityTest.java [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+        performLogIn(email, password);
+        intended(allOf(hasComponent(HomeActivity.class.getName()), hasExtra(USER, user)));
+    }
+
+    private void testEmptyEntryCannotSignUp(int id) {
+        testWrongEntryCannotSignUp(id, "");
+    }
+
+    private void testWrongEntryCannotSignUp(int id, String stringToBeTyped) {
+        fillFields("Balboa", "Bark", "balboa.bark@river.tree", "honeyToast", "honeyToast");
```
Comment (brunnerant) [inline app/src/androidTest/java/ch/epfl/qedit/login/SignUpActivityTest.java:127]: This is not a comment per-se, but I just admire Anthony's inspiration to find funny usernames and passwords. Maybe the great inspirations comes from later in the night.

File: app/src/main/java/ch/epfl/qedit/view/login/LogInActivity.java (hunk 3 above)
Comment (nathan96g) [reply app/src/main/java/ch/epfl/qedit/view/login/LogInActivity.java:166]: right i do this

**Pull request merged**